### DISCUSSION TITLE
Renaming of Modules

### DISCUSSION THREAD

**Discussion by ponceta** [teksi/Home#90] [Q&A] [answered]
To avoid confusion with old QWAT and QGEP versions, we will rename projects supported by the TEKSI community.

Here are proposals for naming :

1. Github organization : 
- [X] TEKSI

2. Modules repositories :

- [x] wastewater
- [x] drinking_water
- [x] district_heating

3. Databases names

- [x] wastewater
- [x] drinking_water
- [x] district_heating

4. Schema prefixes

- [x] tww_od, ww_vl, tww_app, tww_sys, tww_export, tww_network, tww_swmm
- [x] tdw_od, tdw_vl, tdw_app, tdw_sys, tdw_export, tdw_network, tdw_sire, tdw_aquafri
- [x] tdh_od, tdh_vl, tdh_app, tdh_sys, tdh_export, tdh_network

5. Roles

- [x] tww_viewer
- [x] tww_user
- [x] tww_manager
- [x] tww_admin

Any contribution or pro and cons are welcome here

**Comment by jpdupuy** [comment]
schema prefixes and roles : suggest 3 letters instead of 2 : start with the "t" letter for TEKSI, such as tww_od tdw_od

**Reply by ponceta** [reply]
What do you think of a generic approach with module :

- [ ] wastewater module : wwm -> wwm_od, wwm_vl
- [ ] drinking water module : dwm -> dwm_od, dwm_vl
- [ ] distance heating module : dhm -> dhm_od, dhm_vl

**Reply by jpdupuy** [reply]
I would prefer an alphabetic grouping which makes the identification of TEKSI modules in PGADMIN easier.
When you start with the T letter, all TEKSI schemas are grouped together.
We already have dozen of schemas in Postgis.

**Reply by ponceta** [reply]
TEKSI modules are designed to be deployed in separate databases but tww, twm, tdh would also be fine for me too. (just hope we dont change naming too often! ;)

**Comment by sjib** [comment]
General question concerning wastewater: The datamodel we use in TEKSI is VSA-DSS which means **D**aten**s**truktur **S**iedlungsentwässerung that includes not only the Leitungskataster / cadastre des conduites but also the full Werkinformation and beyond that information for the Generelle Entwässerungsplanung (GEP), in French PGEE. 

So Siedlungsentwässerung would be urban drainage (Data Structure urban drainage System (DSUDS)), or in French Assainissement urbain or **E**vacuation des **e**aux urbaines (as in VSA-SD**EE**)

On the datamodel and repository side I would go for this full amount - maybe we can once have a "reduced" module which just offers selected functionality for a user group that does not deal with GEP, but we should not split the datamodel there.

**Reply by sjib** [reply]
So my suggestion: urban_drainage_module: tud_od, tud_vl, etc.

**Reply by ponceta** [reply]
urban drainage sounds a bit like that it is not suited for rural areas?

We would possibly have water distribution and water drainage but this brings us to the samed twd prefixes.

**Comment by urskaufmann** [comment]
humans are creatures of habit - so we will use it the one or the other way. I prefer short. tww is quite common, tud is very unussual for me...  but I see stefans arguments

**Comment by sjib** [comment]
> urban drainage sounds a bit like that it is not suited for rural areas?

Only in English we have this sense - in German it is Siedlungsentwässerung (does not say whether urban or rural), same in French Assainissement des agglomérations

**Comment by sjib** [comment]
@ponceta @jpdupuy @urskaufmann and everyone else:
If I want to continue with model creation we should finish this discussion. Please add your comments till tomorrow evening (27.9.2023).

**Reply by ponceta** [reply, marked as answer]
Please use the tww approach as propose by @jpdupuy and agreed by @urskaufmann. I made the necessary edits in the title post.

Integrated in https://github.com/teksi/Home/wiki/TEKSI-Developer-Guide

**Reply by sjib** [reply]
On the marketing side we should not forget to promote it as "wastewater & gep" module as it is a unique selling point that it not only supports wastewater cadastre but is a tool for the whole GEP maintenance.

![flyer_modules_de](https://github.com/teksi/Home/assets/1265865/32d86e21-2ad0-4ecc-a9a9-1277c9219516)

**Reply by ponceta** [reply]
We could add eau potable et PDDE / trinkwasser XXXX? as the actual water module is able to deliver SIRE and AQUAFRI data to cantons (VD + Fribourg).

https://www.vd.ch/themes/environnement/eaux/eau-potable/distribution-de-leau/plan-directeur-pdde (SIRE)
https://www.fr.ch/energie-agriculture-et-environnement/eau/planification-de-la-distribution-deau-potable (AquaFri)

On the technical naming of modules and databases, I would stick to the strictly minimum but I fully agree on the marketing and tags for referencing.

**Reply by cymed** [reply]
trinkwasser XXXX - Trinkwasser & GWP 
GWP = Generelle Wasserversorgungsplanung

**Comment by ponceta** [comment]
To @ponceta set it as a definitive statement in the developper's guide.

**Comment by ponceta** [comment]
Question is raised from @rbovard why not having :

- tww_ordinary_data **instead of** tww_od
- tww_value_lists **instead of** tww_vl
- tww_application **instead of** tww_app
- tww_system **instead of** tww_sys
- tww_export
- tww_network

There is no restriction on postgresql side except lowercase with undescores.

What we miss today is maybe a good comment on every schema which sets clearly what is accepted in it :

Ex: Comment on tww_od : "Ordinary Data : objects and classes of the datamodel" 
Comment on tww_vl : "Value lists"
Comment on tww_sys : "System administration"
Comment on tww_app : "Application"
Comment on tww_network : "Network following"

**Reply by sjib** [reply]
I agree to having a description on the schemas, but I will not change this again! (I spent now two (!) extra days on adapting to the defined naming). Short schema names help also in reading the code and you see quickly the class and attribute name, that are much more in the center.

**Reply by rbovard** [reply]
In this case it's not consistent with `tww_export` and `tww_network`...

The problem isn't how long it takes to make the change, but whether it's consistent and explicit.

**Reply by ponceta** [reply]
@sjib no aditionnal change is required here in terms of VSA 2020. 

I think it would be easier and more consistent to truncate tww_export to tww_exp and tww_network to tww_net and add useful comments if that's a better option.

The plugins schemas would suffer from such a shortening tdw_sir (sire) and tdw_aqu (aquafri) so consistency is maybe not a friend for user-friendlyness.

If required, postgres schema names could evolve to have a `technical name` and a `common name` designed to be more user friendly?

**Reply by rbovard** [reply]
I think the "core" schema should be using the same form, so:

- tww_export --> tww_exp
- tww_network --> tww_net

And why not prefix plugin schemas? Something like:

- tww_contrib_sire
- tww_contrib_aquafri

I know contrib is not 3 letters but I think it's acceptable, no?

**Reply by ponceta** [reply]
- `tdw_ext_sire`
- `tdw_ext_aquafri`

`ext` meaning extensions?

And this would also group extension due to alphabetic order of schemas

**Reply by rbovard** [reply]
Yes, good proposal.

> And this would also group extension due to alphabetic order of schemas

That was the goal ;)

**Comment by sjib** [comment]
@ponceta I think you started also a list of translations - can you add them to the answer?
Do we then integrate the conclusion in the Developers guide?

**Comment by nicolas-sitylb** [comment]
Hello,

@ponceta :

I think one say "district heating" instead of "distance heating" in English: https://en.wikipedia.org/wiki/District_heating

Even the Swiss Confederation uses this wording: https://www.bfe.admin.ch/bfe/en/home/supply/energy-efficiency/district-heat.html

I would therefore rename this module in consequences.

Thank you for your comprehension.

**Reply by sjib** [reply]
On linguee I can find both:
https://www.linguee.de/deutsch-englisch/search?query=fernw%C3%A4rme
https://www.linguee.de/englisch-deutsch/uebersetzung/long-distance+heating.html

The working group of the SVGW and Thermische Netze Schweiz, that started work for a new release of this data model in January decided to change the name of the Medium from "Fernwärme" to "Wärme" in German. So may be this term needs also adaption in French and English. The planned SIA405 Norm will speak of Wärme from next year on.

**Reply by ponceta** [reply]
I suggest to rename it to the most common occurence which seems to be `district heating`

**Reply by sjib** [reply]
@ponceta Can you adjust the Developer guide if this is final: https://github.com/teksi/Home/wiki/TEKSI-Developer-Guide#modules-repositories-

**Reply by sjib** [reply]
@jpdupuy Is the suggested change ok for you?

**Comment by jpdupuy** [comment]
@sjib yes, fine with me
